Conv1d (#2103)

YushaArif99 opened the pull request.

rush2406 commented:
@YushaArif99 
Thanks for contributing! please check the comments and re-request review

YushaArif99 commented:
> @YushaArif99 Thanks for contributing! please check the comments and re-request review when done. Please also correct the lint issues as seen in the `lint / check formatting` in this PR.

Thanks for the review. I'll make the changes requested!

rush2406 commented on ivy/functional/ivy/layers.py:
```diff
@@ -500,7 +500,51 @@ def conv1d(
+With :code:`ivy.NativeArray` input:
+
+>>> x = ivy.native_array(ivy.random_normal(0, 1,[1, 5, 3])) #NWC
+>>> filters = ivy.native_array(ivy.random_normal(0, 1,  [3, 3, 1])) #WIO
```
same issue `1<3`

rush2406 commented on ivy/functional/ivy/layers.py:
```diff
@@ -500,7 +500,51 @@ def conv1d(
+
+ >>> x = ivy.Container(a = ivy.array(ivy.random_normal(0,1, [1, 10, 3])) , \
+                       b = ivy.array(ivy.random_normal(0,1,[1,20, 3 ])))
+ >>> filters = ivy.array(ivy.random_normal(0,1,[5, 3, 2]))
```
same issue here `1<5`

rush2406 commented:
@YushaArif99 thanks for doing the changes, could you please try to address the comments and update the examples accordingly? I believe the issue is because of the problem with filter size being greater than the input size.
Secondly, the `numpy-docstring` tests are failing because of some formatting issue, could you please fix them as well.

YushaArif99 commented on ivy/array/layers.py:
```diff
@@ -59,3 +59,64 @@ def depthwise_conv2d(
+        Examples
+        --------
+        >>> x = ivy.array(ivy.random_normal(0, 1, [2, 12, 4]))  # NWC
+        >>> filters = ivy.array(ivy.random_normal(0, 1, [6, 4, 2]))  # WIO (I == C)
```
yeah sure i'll have a look

YushaArif99 replied:
I just reran the example. It worked perfectly fine for me.
Why is it giving an error when you're running?

YushaArif99 replied:
The 2<6 shouldn't be a problem as far as I can tell since the 2 represents the batch-size. As far as the shapes are concerned, The **C** parameter of the input (`NWC`) should match the **I** parameter of the filter(`WIO`).

YushaArif99 commented:
Thanks for the review @rush2406. I'm not sure why the docstring tests are failing since they're all running perfectly on my personal machine. However, I'll still change all the examples to ensure **N >= W.**
As for the `numpy-style docstring,` I'll look into it and resolve any errors.

YushaArif99 commented:
I have changed the examples. I don't think there was an issue with the examples themselves. Rather, the issue might be with the tests that have been constructed. Nonetheless, after looking through the docstring examples of other methods  (eg: `depthwise_conv2d`), I've realized that the examples constructed were rather quite simple. Hence, I have also made changes whereby I've set the input and output channels to be of the same shape. Also, I have set the value of **N=1** for all examples. Let me know if the examples are still giving an error

YushaArif99 commented:
After looking at the `test-docstrings` file, I realized that even after making the above changes, the docstring tests were still failing. Upon closer inspection at the contents of that file, I've figured that has nothing to do with incorrect shapes to the function but is dependent on the format of the output in the docstring. I guess the tests have been designed to expect the output to be in a single line and not properly formatted(i.e with spaces and indentations etc). Thus, I have rewritten the examples keeping the above point in mind. Let me know if there's still some issue.

rush2406 reviewed (changes requested):
Looks great! There's a minor issue with the `lint tests`. I'm pasting the specific error for your reference: `ivy/container/layers.py:176:89: E501 line too long (89 > 88 characters)`. Make sure the lint test passes after this change. Please request a review when done and it should be good to merge!

YushaArif99 commented:
@rush2406 I've corrected all the test cases. The `test-docstring` file is now running without error for the **conv1d** function. I've also corrected the `numpy-style-docstring` errors as well the` lint-formatting` errors. Let me know if there are any more issues.

rush2406 commented:
Thanks @YushaArif99! I have merged the PR

YushaArif99 commented:
Thanks alot @rush2406!. Are we allowed to work on some other open tasks in the meantime?
